base/special/j0f.jl

```diff
@@ -0,0 +1,331 @@
+            z = invsqrtpi * (u * cc - v * ss) / sqrt(x)
+        end
+        return z
+    end
+    if ix < 0x3b000000 # |x| < 2^-9
+        if huge + x > 1f0 # raise inexact if x != 0
+            if ix < 0x39800000
+                return 1f0 # |x| < 2^-12
+            else
+                return 1f0 - (x * x) / 4
+            end
+        end
+    end
+    z = x * x
+    r = z * (R02 + z * (R03 + z * (R04 + z * R05)))
+    s = 1f0 + z * (S01 + z * (S02 + z * (S03 + z * S04)))
+    if ix < 0x3F800000 # |x| < 1.00
+        return 1f0 + z * (-0.25f0 + r / s)
+    else
+        u = 0.5f0 * x
+        return (1f0 + u) * (1f0 - u) + z * (r / s)
+    end
+end
+
+const u00 = -7.3804296553f-02 # 0xbd9726b5
+const u01 =  1.7666645348f-01 # 0x3e34e80d
+const u02 = -1.3818567619f-02 # 0xbc626746
+const u03 =  3.4745343146f-04 # 0x39b62a69
+const u04 = -3.8140706238f-06 # 0xb67ff53c
+const u05 =  1.9559013964f-08 # 0x32a802ba
+const u06 = -3.9820518410f-11 # 0xae2f21eb
+const v01 =  1.2730483897f-02 # 0x3c509385
+const v02 =  7.6006865129f-05 # 0x389f65e0
+const v03 =  2.5915085189f-07 # 0x348b216c
+const v04 =  4.4111031494f-10 # 0x2ff280c2
+
+function ieee754_y0f(x::Float32)
+    hx = reinterpret(Int32, x)
+    ix = 0x7fffffff & hx
+    # y0(NaN) is NaN, y0(-Inf) is NaN, y0(Inf) is 0
+    ix >= 0x7f800000 && return 1f0 / (x + x * x)
+    ix == 0 && return -Inf32
+    # hx < 0 && return NaN32
+    # use (x < 0) so that llvm can infer that x >= 0 afterwards
+    x < 0 && return NaN32
+    if ix >= 0x40000000 # |x| >= 2.0
+        # y0(x) = sqrt(2 / (π * x)) * (p0(x) * sin(x0) + q0(x) * cos(x0))
+        # where x0 = x - π / 4
+        #      Better formula:
+        #              cos(x0) = cos(x)cos(π / 4) + sin(x)sin(π / 4)
+        #                      =  1/√2 * (sin(x) + cos(x))
+        #              sin(x0) = sin(x)cos(3π / 4) - cos(x)sin(3π / 4)
+        #                      =  1/√2 * (sin(x) - cos(x))
+        # To avoid cancellation, use
+        #              sin(x) ± cos(x) = -cos(2x) / (sin(x) ∓ cos(x))
+        # to compute the worse one.
+        s = sin_unchecked(x)
+        c = cos_unchecked(x)
+        ss = s - c
+        cc = s + c
+        # j0(x) = 1 / √π * (P(0, x) * cc - Q(0, x) * ss) / sqrt(x)
+        # y0(x) = 1 / √π * (P(0, x) * ss + Q(0, x) * cc) / sqrt(x)
+        if ix < 0x7f000000 # make sure x + x not overflow
+            z = -cos_unchecked(x + x)
+            if s * c < 0f0
+                cc = z / ss
+            else
+                ss = z / cc
+            end
+        end
+        if ix > 0x58000000 # |x| > 2^49
+            z = (invsqrtpi * ss) / sqrt(x)
+        else
+            u = pzerof(x)
+            v = qzerof(x)
+            z = invsqrtpi * (u * ss + v * cc) / sqrt(x)
+        end
+        return z
+    end
+    if ix <= 0x39000000 # x < 2^-13
+        return u00 + tpi * log_unchecked(x)
+    end
+    z = x * x
+    u = u00 + z * (u01 + z * (u02 + z * (u03 + z * (u04 + z * (u05 + z * u06)))))
```

Comment:
I was a little too lazy to check if it is defined early enough in boot strap and wasn't aware that `@evalpoly` is actually using `muladd`. Yeah, I think it's a good idea and I'll replace it.